spinalcordtoolbox/scripts/sct_label_vertebrae.py

```diff
@@ -415,11 +436,16 @@ def main(argv=None):
             img.data = laplacian(img.data, sigmas)
             img.save()
 
-
         # detect vertebral levels on straight spinal cord
         init_disc[1] = init_disc[1] - 1
-        vertebral_detection('data_straightr.nii', 'segmentation_straight.nii', contrast, param, init_disc=init_disc,
-                            verbose=verbose, path_template=path_template, path_output=path_output, scale_dist=scale_dist)
+        if arguments.method == 'TM':
+            vertebral_detection('data_straightr.nii', 'segmentation_straight.nii', contrast, param, init_disc=init_disc,
+                                verbose=verbose, path_template=path_template, path_output=path_output,
+                                scale_dist=scale_dist)
+        elif arguments.method == 'DL':
+            deep_method.vertebral_detection('data_straightr.nii', 'segmentation_straight.nii', contrast, param, init_disc=init_disc,
+                                            verbose=verbose, path_template=path_template, path_output=path_output,
+                                            scale_dist=scale_dist)
```

Comment:
This is the section of code where the straightened input data is passed to the vertebral detection function.